django/forms/utils.py

```diff
@@ -41,53 +43,91 @@ def flatatt(attrs):
     )
 
 
-@html_safe
-class ErrorDict(dict):
+class Renderable:
+    def get_context(self):
+        raise NotImplementedError('Must be overridden by subclasses.')
+
+    def render(self, template_name=None, context=None, renderer=None):
+        return mark_safe((renderer or self.renderer).render(
+            template_name or self.template_name,
+            context or self.get_context(),
+        ))
+
+    def __str__(self):
+        return self.render()
+
+    def __html__(self):
+        return self.render()
```

Comment:
I think these _could_ be:
```
__str__ = render
__html__ = render
```
Avoiding +1 depth (and it's faster).

Off the top of my head, the only reasons _not_ to would be:
- `x.__str__` would report `<bound method X.render of ...>` instead of `<bound method X.__str__ of ...>`
- you _could_ technically manually call `x.__str__(template_name='a')` but I mean, who's doing that?

Discuss amongst yourselves whether to change it, I don't think it really matters much :)

Reply:
I've changed this. 

One thing for a follow up is the `@html_safe` decorator of `Media`, `BoundField` and `BoundWidget`, could they benefit from a similar sort of treatment? Not looked too closely, but I suspect there's multiple `mark_safe` calls in addition to the note you raised here.

Reply:
Whilst I haven't yet looked at those ones you've mentioned (will try and do so shortly), _personally_, I wouldn't stress too much over there being _too many_ calls to `mark_safe`. Wrapping over the same `SafeString` multiple times is fine, if unnecessary, and first priority is better safe than sorry. Unwrapping a quagmire of nested safety decorators is a concern which _could_ be deferred until later, I think?